### PR TITLE
Implements an initial version of the Plugin interface.

### PR DESCRIPTION
**Pull request by diegoreymendez**
### Description:

This PR implements an initial version of our Plugin interface, to make it easier to group code up, and to avoid having to pass processors all around our code.

### Details:

This PR: 

- Provides an initial version of our [Plugin interface](https://github.com/wordpress-mobile/AztecEditor-iOS/compare/try/implement-plugins-interface?expand=1#diff-6cc1c378e97a70aa0c8fd0ca4678bb96R3).
- Removes an extension that was only used [in one place](https://github.com/wordpress-mobile/AztecEditor-iOS/compare/try/implement-plugins-interface?expand=1#diff-38ed2d0ecd5bad90dfb68e193ea81127L6).  This will also allow us in the future to have a single instance of our parser / serializers.
- [Adds methods](https://github.com/wordpress-mobile/AztecEditor-iOS/compare/try/implement-plugins-interface?expand=1#diff-6cc1c378e97a70aa0c8fd0ca4678bb96R21) to make it easy to do processing with our plugins.
- Renames some processors [to better describe what they do](https://github.com/wordpress-mobile/AztecEditor-iOS/compare/try/implement-plugins-interface?expand=1#diff-714fdd29f4d9e85e2db73e1f7c69c8bfR4).
- Implements [a Plugin loading interface](https://github.com/wordpress-mobile/AztecEditor-iOS/compare/try/implement-plugins-interface?expand=1#diff-8d8015f3f021c0252c1569e4965dce80R197) in `TextView`.
- [Introduces a WordPress plugin](https://github.com/wordpress-mobile/AztecEditor-iOS/compare/try/implement-plugins-interface?expand=1#diff-668aa2c6bb496dbe292a6e49cd3d563fR3) containing all of the Calypso logic, and soon the Gutenberg logic as well.
- We had [duplicated video processors](https://github.com/wordpress-mobile/AztecEditor-iOS/pull/954/files#diff-5a57f12cbf3d1eb4025c14b8174187afL4) apparently, so I removed some of them.  @SergioEstevao since you probably know best about these, can you double check I didn't delete anything important?

### Testing:

1. Run the unit tests.
2. Play with the editor.
3. WPiOS integration branch: `try/integrate-latest-aztec`

### CODE REVIEW

**Comment by diegoreymendez** [comment]
Fyi, @koke & @hypest.

While not strictly related to the compatibility work, this was becoming a bit necessary to avoid passing a lot of components all around the place.

File: Aztec/Classes/Plugin/Plugin.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import Foundation
+
+open class Plugin {
```
Comment (SergioEstevao) [inline Aztec/Classes/Plugin/Plugin.swift]: Will be good to add documentation to the class and methods to explain what is a plugin and how to use, and where they act on the Aztec parsing process.

File: Aztec/Classes/Plugin/Plugin.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    open func getInputHTMLTreeProcessor() -> HTMLTreeProcessor? { return nil }
+    open func getOutputHTMLTreeProcessor() -> HTMLTreeProcessor? { return nil }
+    open func getElementConverter() -> ElementConverter? { return nil }
+
+    public static func ==(lhs: Plugin, rhs: Plugin) -> Bool {
+        return type(of: lhs) == type(of: rhs)
+    }
+}
+
+extension Array where Element: Plugin {
```
Comment (SergioEstevao) [inline Aztec/Classes/Plugin/Plugin.swift]: IMHO opinion I think this are better of has methods on the TextStorage class than extension to Array. 

Like the extension on NSAtributtedString that you removed above I don't see this being used in another places.

**Comment by SergioEstevao** [comment]
>We had duplicated video processors apparently, so I removed some of them. @SergioEstevao since you probably know best about these, can you double check I didn't delete anything important?

The processor that you are deleting where the ones used to test the processor on the pod before we merged in the processors from the main app. It's ok to delete them now.

**Comment by diegoreymendez** [comment]
@SergioEstevao - Changes are up, and the WPiOS integration branch is provided as well.

**Comment by diegoreymendez** [comment]
Thank you!

**Pull request merged**